packages/manifold/contracts/gachaclaims/GachaLazyClaim.sol

```diff
@@ -0,0 +1,92 @@
+// SPDX-License-Identifier: MIT
+// solhint-disable reason-string
+pragma solidity ^0.8.0;
+
+import "@manifoldxyz/libraries-solidity/contracts/access/AdminControl.sol";
+
+import "./IGachaLazyClaim.sol";
+
+/**
+ * @title Gacha Lazy Claim
+ * @author manifold.xyz
+ */
+abstract contract GachaLazyClaim is IGachaLazyClaim, AdminControl {
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  string internal constant ARWEAVE_PREFIX = "https://arweave.net/";
+  string internal constant IPFS_PREFIX = "ipfs://";
+  address internal _signer;
+
+  uint256 internal constant MAX_UINT_24 = 0xffffff;
+  uint256 internal constant MAX_UINT_32 = 0xffffffff;
+  uint256 internal constant MAX_UINT_56 = 0xffffffffffffff;
+  uint256 internal constant MAX_UINT_80 = 0xffffffffffffffffff;
+  uint256 internal constant MAX_UINT_256 = 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff;
+  address internal constant ADDRESS_ZERO = 0x0000000000000000000000000000000000000000;
+
+  uint256 public constant MINT_FEE = 500000000000000;
+
+  // { contractAddress => { instanceId => { walletAddress => reservedMints } } }
+  mapping(address => mapping(uint256 => mapping(address => uint32))) internal _reservedMintsPerWallet;
```

Comment:
Only have ONE mapping.  With a single struct called WalletMintInfo

Right now, each person will have two storage slots.  Using a map with a single struct, you can use only one storage slot.